lib/ts/axios.ts

```diff
@@ -134,6 +134,9 @@ export async function interceptorFunctionRequestFulfilled(config: AxiosRequestCo
     const accessToken = getTokenForHeaderAuth("access");
     if (accessToken !== undefined) {
         if (configWithAntiCsrf.headers!.authorization === `Bearer ${accessToken}`) {
+            // We are ignoring the Authorization header set by the user in this case, because it would cause issues
+            // If we do not ignore this, then this header would be used even if the request is being retried after a refresh, even though it contains an outdated access token.
+            // This causes an infinite refresh loop.
             logDebugMessage(
                 "interceptorFunctionRequestFulfilled: Removing Authorization from user provided headers because it contains our access token"
             );
```

Comment:
I think calling `await setAuthorizationHeaderIfRequired(configWithAntiCsrf);` below is wrong?

Reply:
we should definitely call it in `interceptorFunctionRequestFulfilled` otherwise the first request would always fail. After checking it out:
- The tests didn't cover this because `addXMLHttpRequestInterceptor` disabled the axios interceptor. I've changed this.
- This check should be done before going into the refresh loop in `doRequest`